packages/react-native/ReactCommon/react/renderer/mounting/ShadowTree.cpp

```diff
@@ -374,11 +379,6 @@ CommitStatus ShadowTree::tryCommit(
 
     auto newRevisionNumber = oldRevision.number + 1;
 
-    if (!newRootShadowNode ||
```

Comment:
I think it's ok to remove the `!newRootShadowNode` check from here because we already checked it in the new block you added, but I think we should keep the rest. Layout is relatively slow so it could happen that the commit is requested to yield during layout, and we should still be able to cancel then. Changing this behavior without testing its impact would be risky.

Reply:
sure thing.

By the way, what do you think about restoring that block with moving it just after getting a lock and before checking for revision numbers' mismatch? It may allow developers to cancel commit instead of getting failure on specific cases and shouldn't cause any behaviour changes

Reply:
I've pushed suggested changes, @rubennorte give me a shout about your opinion on that approach

Reply:
Yeah, I think that makes sense! thanks!